rig/machine_control/scp_connection.py

```diff
@@ -0,0 +1,152 @@
+"""A stop-and-wait blocking implementation of the SCP protocol.
+"""
+import socket
+from . import packets
+
+
+class SCPConnection(object):
+    """Implements the SCP protocol for communicating with a SpiNNaker machine.
+    """
+    error_codes = {}
+
+    def __init__(self, spinnaker_host, n_tries=5, timeout=0.5):
+        """Create a new communicator to handle control of the given SpiNNaker
+        host.
```

Comment:
"Create a new communicator to handle control of the SpiNNaker chip with the supplied hostname."